test/servinge2e/servicemesh_test.go

```diff
@@ -85,11 +125,39 @@ func TestKsvcWithServiceMeshSidecar(t *testing.T) {
 				return
 			}
 
+			serviceURL := service.Status.URL.URL()
+
+			// For cluster-local ksvc, we deploy an "HTTP proxy" service, and request that one instead
+			if service.GetLabels()[serving.VisibilityLabelKey] == serving.VisibilityClusterLocal {
+				// Deploy an "HTTP proxy" towards the ksvc (using an httpproxy image from knative-serving testsuite)
+				httpProxy, err := caCtx.Clients.Serving.ServingV1().Services(serviceMeshTestNamespaceName).Create(
+					httpProxyService(scenario.name+"-proxy", service.Status.URL.Host))
+				if err != nil {
+					t.Errorf("error creating ksvc: %v", err)
```

Comment:
Would it be possible to use t.Fatalf instead of t.Errorf plus return?

Reply:
We want the other scenarios to continue in case of error.. (and I guess we'd lose that if we did Fatalf, but I am not sure tbh.. )

Reply:
Hmm. It should end just the current test. The Fatalf is equivalent to Logf followed by FailNow. And FailNow docs: https://golang.org/src/testing/testing.go?s=23520:23546#L682

Reply:
FailNow would call runtime.goexit , so this would skip all the other TestKsvcWithServiceMeshSidecar scenarios...

Reply:
(we run all TestKsvcWithServiceMeshSidecar scenarios in a single goroutine)

Reply:
hmm..or don't we? what exactly does t.Run  do  if we are not running parallel...

Reply:
ok, maybe we do then... let's try it then...

Reply:
I don't know :-D